Modules/Module_Import/NII2BIDS/xASL_bids_PhoenixProtocolAnalyzer.m

```diff
@@ -14,198 +14,256 @@
-	%TODO - does it really work for other sequences?
-    if ~isempty(sourcePar.alTI0) && ~isempty(sourcePar.alTI2)
-        if sourcePar.alTI0~=100000 && sourcePar.alTI2~=7000000
-            bidsPar.ScanType = 'ASL';
-        elseif sourcePar.alTI0==100000 && sourcePar.alTI2==7000000
-            bidsPar.ScanType = 'PseudoM0';
-        end
-	end
-    
-	if isfield(sourcePar,'sProtConsistencyInfotBaselineString') && ~isempty(sourcePar.sProtConsistencyInfotBaselineString)
-		bidsPar.SoftwareVersions = strrep(sourcePar.sProtConsistencyInfotBaselineString,'"','');
+	bSequenceIdentified = false; % Try to identify different sequences and in case this doesn't work we rollback to the defaults
+
+	%% 3. Reading sequence VEPCASL
+	if ~bSequenceIdentified && ~isempty(regexp(sourcePar.tSequenceFileName,'VEPCASL', 'once'))
+		if ~isempty(regexp(sourcePar.tSequenceFileName,'ep2d_VEPCASL', 'once'))
+			% Basic check
+			if ~isempty(sourcePar.sWipMemBlockalFree9) &&  ~isempty(sourcePar.sWipMemBlockalFree11) 
+				% All basic passes checked, we can proceed to parameter parsing
+				bSequenceIdentified = true;
+
+				% Load the required parameters
+				bidsPar.PostLabelingDelay = sourcePar.sWipMemBlockalFree11 / 1000;
+				bidsPar.LabelingDuration = sourcePar.sWipMemBlockalFree11 / 1000;
+
+				if ~isempty(sourcePar.sWipMemBlockalFree3) && sourcePar.sWipMemBlockalFree3 == 2
+					bidsPar.BackgroundSuppression = true;
+				end
+
+				if ~isempty(sourcePar.sWipMemBlockalFree4)
+					bidsPar.LabelingPulseFlipAngle = sourcePar.sWipMemBlockalFree4;
+				end
+
+				if ~isempty(sourcePar.sWipMemBlockalFree5)
+					bidsPar.LabelingPulseDuration = sourcePar.sWipMemBlockalFree5 / 1000;
+					if ~isempty(sourcePar.sWipMemBlockalFree6)
+						bidsPar.LabelingPulseInterval = sourcePar.sWipMemBlockalFree6 / 1000 + bidsPar.LabelingPulseDuration;
+					end
+				end
+
+				if ~isempty(sourcePar.sWipMemBlockadFree0)
+					bidsPar.LabelingPulseAverageGradient = sourcePar.sWipMemBlockadFree0;
+				end
+
+				if ~isempty(sourcePar.sWipMemBlockadFree1)
+					bidsPar.LabelingPulseMaximumGradient = sourcePar.sWipMemBlockadFree1;
+				end
+			else
+				warning('Missing important parameters in Phoenix for Siemens VEPCASL sequence');
+			end
+		else
+			% An unknown version detected, skipping to the default
+			warning('An unknown version of Siemens VEPCASL sequence detected');
+		end
 	end
-	
-	% N4_VD13D and N4_VE11C have also slab_thickness_mm parameter defined and it is unclear if this is imaging FOV or labeling slab thickness...
-	if isfield(bidsPar,'SoftwareVersions') &&...
-			(~isempty(regexpi(bidsPar.SoftwareVersions,'N4_VB15A'))||...
-			 ~isempty(regexpi(bidsPar.SoftwareVersions,'N4_VB17A'))||...
-			 ~isempty(regexpi(bidsPar.SoftwareVersions,'N4_VB19A')))
-		if isfield(sourcePar,'sGroupArraysPSatdGap') && ~isempty(sourcePar.sGroupArraysPSatdGap)
-			bidsPar.LabelingDistance = sourcePar.sGroupArraysPSatdGap;
+
+	%% 4. Reading a default sequence
+	if ~bSequenceIdentified
+		%TODO
```

Comment:
We used PM (pro memoria) previously. I don't care what we do, but it would be nice to have a searchable single word we use :)

Reply:
Actually, this was there before. And is not associated with any issue. 
I'd still prefer TODO as this is easier searcheable than PM. Unless we of course have a specific issue, then PM #486 
OK?